### ISSUE TITLE
Remove models with same set of field. 

### ISSUE THREAD

**Issue by gk-patel** [koxudaxi/datamodel-code-generator#170]
It would be good if we can identify that a similar model exists and use that model instead of creating a new model, especially when the FIELDS and DATA-TYPE are same. 

For example, when my data is as follows, 
```
{
    "Arm Right": {
        "Joint 1": 5,
        "Joint 2": 3, 
        "Joint 3": 66
    }, 
    "Arm Left": {
        "Joint 1": 55,
        "Joint 2": 13, 
        "Joint 3": 6
    }
}
```
we get the following output, 
```
class ArmRight(BaseModel):
    Joint_1: int = Field(..., alias='Joint 1')
    Joint_2: int = Field(..., alias='Joint 2')
    Joint_3: int = Field(..., alias='Joint 3')

class ArmLeft(BaseModel):
    Joint_1: int = Field(..., alias='Joint 1')
    Joint_2: int = Field(..., alias='Joint 2')
    Joint_3: int = Field(..., alias='Joint 3')

class Model(BaseModel):
    Arm_Right: ArmRight = Field(..., alias='Arm Right')
    Arm_Left: ArmLeft = Field(..., alias='Arm Left')
```
**Describe the solution you'd like**
I think it would be good if we can detect that models for ArmLeft and ArmRight are the same, and then to just use one of these models. For example, 

```
class ArmRight (BaseModel):
    Joint_1: int = Field(..., alias='Joint 1')
    Joint_2: int = Field(..., alias='Joint 2')
    Joint_3: int = Field(..., alias='Joint 3')

class Model(BaseModel):
    Arm_Right: ArmRight = Field(..., alias='Arm Right')
    Arm_Left: ArmRight = Field(..., alias='Arm Left')
```

**Describe alternatives you've considered**
Currently, I just delete manually. But, we have a big system with many similar structures in JSON and many time we need to manually delete about 1000 models. :( 

Thank you very much for this excellent tool.

**Comment by koxudaxi** [comment]
@gk-patel 
Thank you for creating this issue.
I think it's a good idea.
The behavior may have to be an option of the command line :thinking: 

I will implement this feature after remove bugs.

**Comment by koxudaxi** [comment]
@gk-patel 
I have released a new version as 0.6.15.
The version has the `--reuse-model` option that you want.

I put an example.

Input
https://github.com/koxudaxi/datamodel-code-generator/blob/681ca87e7ee0c25a9e83dfe2a4cc9916083f1229/tests/data/json/duplicate_models.json#L1-L15

Output
https://github.com/koxudaxi/datamodel-code-generator/blob/681ca87e7ee0c25a9e83dfe2a4cc9916083f1229/tests/data/expected/main/main_json_resuse_model/output.py#L10-L29

I'm sorry, My work is too late.

**Comment by gk-patel** [comment]
@koxudaxi  No worries man, looks nice. 

Although, the following would have been much more cooler, 

```
...
class ArmLeft(ArmRight): 
   pass 
...
```

**Comment by koxudaxi** [comment]
@gk-patel 
No problem, It's a good idea.
I can implement the changes easily.

But, I found a problem with a case.
The code generator generates the `Enum` class from JSONSchema/OpenAPI.
Enum class can't be extended.
```
>> class B(A): pass
TypeError: Cannot extend enumerations
```

We should assign to another value. or we don't touch duplicated enum 🤔

**Comment by gk-patel** [comment]
Well, IMHO the easiest way would be to check if the content of enums is exactly the same, 

- if it is exactly the same, just use the same name everywhere (so in that case, dont create class B)
- if it not exactly the same, the create a seperate enum without inheritance, (so in that case, create class A and B) 

Do you have any arguments against this idea ?

**Comment by koxudaxi** [comment]
@gk-patel 
OK. I released a new version as `0.6.17`
This version has changed your idea.

https://github.com/koxudaxi/datamodel-code-generator/blob/ada9b19d4c51b22ac4d4a872794416ad71f329cd/tests/data/expected/main/main_json_reuse_model/output.py#L16-L26

The enum will reuse the same model. A duplicate model will be created too. Because some models may need the enum from other modules. 
https://github.com/koxudaxi/datamodel-code-generator/blob/ada9b19d4c51b22ac4d4a872794416ad71f329cd/tests/data/expected/main/main_json_reuse_enum/output.py#L13-L28
